spyder/utils/tests/test_encoding.py

```diff
@@ -10,12 +10,46 @@
 import stat
 
 from spyder.utils.encoding import is_text_file, get_coding, write
-from spyder.py3compat import to_text_string
+from spyder.py3compat import to_text_string, PY2
+
+if PY2:
+    import pathlib2 as pathlib
+else:
+    import pathlib
 
 __location__ = os.path.realpath(os.path.join(os.getcwd(),
                                              os.path.dirname(__file__)))
 
 
+def test_symlinks(tmpdir):
+    """
+    Check that modifing symlinks files changes source file and keeps symlinks.
+    """
+    base_dir = tmpdir.mkdir("symlinks")
+    base_file = base_dir.join("symlinks_text.txt")
+    base_file_path = to_text_string(base_file)
+
+    # Write base file
+    write("Some text for symlink", base_file_path)
+
+    # Create symlink
+    symlink_file = pathlib.Path(base_dir.join(
+        'link-to-symlinks_text.txt'))
+    symlink_file.symlink_to(base_file_path)
+    symlink_file_path = to_text_string(symlink_file)
+
+    # Assert the symlink was created
+    assert os.path.islink(symlink_file_path)
+
+    # Write using the symlink
+    enconding = write("New text for symlink", symlink_file_path)
+
+    # Assert symlink is valid and contents of the file
+    assert os.path.islink(symlink_file_path)
+    assert base_file.read_text(enconding) == symlink_file.read_text(enconding)
+    assert symlink_file.read_text(enconding) == 'New text for symlink'
```

Comment:
```suggestion
    text = write("New text for symlink", symlink_file_path)

    # Assert symlink is valid and contents of the file
    assert os.path.islink(symlink_file_path)
    assert base_file.read_text(text) == symlink_file.read_text(text)
    assert symlink_file.read_text(text) == 'New text for symlink'
```

Reply:
@ccordoba12 I called the variable `encoding` since that's what the `write` function returns and also that's the first argument of [`read_text`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text). Maybe I should change the call to read text from `read_text(encoding)` to `read_text(encoding=encoding)` for clarity?

Reply:
Ok, no problem. Still, there was a spelling error because you wrote `enconding` instead of `encoding` as the variable defined above. I updated my suggestion to fix that.